src/stream/src/executor/backfill/arrangement_backfill.rs

```diff
@@ -504,6 +521,26 @@ where
         }
     }
 
+    #[try_stream(ok = Option<(VirtualNode, StreamChunk)>, error = StreamExecutorError)]
+    async fn make_snapshot_stream<'a>(
+        upstream_table: &'a ReplicatedStateTable<S, SD>,
+        backfill_state: BackfillState,
+        builders: &'a mut [DataChunkBuilder],
+        paused: bool,
+    ) {
+        if paused {
+            #[for_await]
+            for _ in tokio_stream::pending() {
+                yield None;
+            }
+        } else {
+            #[for_await]
+            for r in Self::snapshot_read_per_vnode(upstream_table, backfill_state, builders) {
+                yield r?;
+            }
+        }
+    }
+
```

Comment:
Is it essentially an [`Either`](https://docs.rs/futures/0.3.30/futures/future/enum.Either.html#impl-Stream-for-Either%3CA,+B%3E) stream of `pending` or `snapshot`? So that we don't need the `yield` syntax.

Reply:
Not really. The purpose of this function is to unify the types for `tokio_stream::pending` and `snapshot_read_per_vnode`.

They should both yield the same items, in terms of their types.

The async behaviour of the streams are different however.
If `paused`, we will never return any records from the snapshot read side.

If not `paused`, we will return records from snapshot.

Reply:
Can see https://github.com/risingwavelabs/risingwave/pull/14745#issuecomment-1909419725

Reply:
I see. I mean something like this:

```rust
if paused {
  Either::Left(futures::stream::pending())
} else {
  Either::Right(Self::snapshot_read_per_vnode(upstream_table, backfill_state, builders))
}
```

Reply:
I don't think the concrete types will align for that. `Self::snapshot_read_per_vnode` will have a derived type, which will mismatch the type from `futures::stream::pending()`.

Reply:
`Either` can have different type on each arm. 😄